### PR TITLE
Tech 7284 upgrade ruby and bundler

### PR DESCRIPTION
**Pull request by gwynethallwright**

### CODE REVIEW

**Comment by gwynethallwright** [comment]
Let me know if you'd like to see the prerelease tested against a web build first.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -33,7 +33,42 @@ jobs:
...
+          MYSQL_ROOT_PASSWORD: rootpassword
+        ports:
+        - 3306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: ['3.0', 3.1]
+        gemfile:
+        - gemfiles/rails_6.gemfile
```
Comment (jebentier) [inline .github/workflows/test.yml]: There's no need to split between ruby 2 and ruby 3 like this just to exclude testing of rails 5 on ruby 3+. The `matrix` syntax allows the ability to do an `exclude` section so the matrix definition would look like the following:

```
matrix:
  ruby: [2.5, 2.6, 2.7, '3.0', 3.1]
  gemfile:
  - Gemfile
  - gemfiles/rails_5.gemfile
  - gemfiles/rails_6.gemfile
  exclude:
  - ruby: '3.0'
    gemfile: gemfiles/rails_5.gemfile
  - ruby: 3.1
    gemfile: gemfiles/rails_5.gemfile
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,10 @@ Inspired by [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 Note: this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [2.0.1] - Unreleased
+- Upgraded to Ruby 2.7 and Bundler 2.
+- Added support for Ruby 3.0 and 3.1.
+
```
Comment (jebentier) [inline CHANGELOG.md]: No need to include project development environment changes as gem updates.
```suggestion
## [2.0.1] - Unreleased
### Added
- Added support for Ruby 3.0 and 3.1.

```

**Review by jebentier — CHANGES_REQUESTED** [review]
It would be good to see at least a proof that the gem is passing in web before merging as a QA item.

File: .github/workflows/test.yml (hunk 1 above)
Comment (gwynethallwright) [reply .github/workflows/test.yml]: Neat! Thanks.

**Comment by gwynethallwright** [comment]
> It would be good to see at least a proof that the gem is passing in web before merging as a QA item.

Web build here: https://buildkite.com/invoca/web/builds/25494.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 name: ActiveRecord MySQL Enum
 on: [push]
 jobs:
-  test:
+  test-ruby:
```
Comment (jebentier) [inline .github/workflows/test.yml]: can we set this back to just `test`

**Review by jebentier — APPROVED** [review]
Everything looks good to me, thanks for linking the web build to prove out that this all works and is non-breaking. Once the minor change is made to the github workflow, feel free to merge.

**Review by jebentier — APPROVED** [review]

**Pull request merged**